### PR TITLE
add fluentd in contrail services

### PR DESCRIPTION
**Pull request by k-ayache**

### CODE REVIEW

File: pkgs/docker-images/default.nix [hunk 1]
```diff
@@ -45,20 +45,76 @@ in
 
   contrailApi = buildContrailImageWithPerp {
     name = "opencontrail/api";
-    command = "${contrail32Cw.api}/bin/contrail-api --conf_file /etc/contrail/contrail-api.conf";
+    command = "${pkgs.bash}/bin/bash -c '${contrail32Cw.api}/bin/contrail-api --conf_file /etc/contrail/contrail-api.conf 2>&1'";
```
Comment (nlewo) [inline pkgs/docker-images/default.nix]: Could you do this in the genPerpRcMain function?

File: pkgs/docker-images/default.nix [hunk 2]
```diff
@@ -45,20 +45,76 @@ in
...
     preStartScript = my_ip + ''
       consul-template-wrapper -- -once \
         -template="${config.contrail.api}:/etc/contrail/contrail-api.conf"
     '';
+    fluentd = {
+      source = {
+        type = "stdout";
+        time_key = "time";
```
Comment (eonpatapon) [inline pkgs/docker-images/default.nix]: Wondering what this option do. Couldn't find it in the docs

File: pkgs/docker-images/default.nix (hunk 1 above)
Comment (eonpatapon) [reply pkgs/docker-images/default.nix]: Maybe just in buildContrailImageWithPerp since it's a contrail workaround

Comment (k-ayache) [reply pkgs/docker-images/default.nix]: I dont think so , some services like control, collector are not concerned by this redirection. they are on stdout by default

File: pkgs/docker-images/config/contrail.nix [hunk 3]
```diff
@@ -7,7 +7,42 @@ let
   '' + headers
      + pkgs.lib.generators.toINI {} conf;
 
-  logConfig = service: {
+  logging = pkgs.writeTextFile {
```
[the remark this answers is not in the document]
Comment (k-ayache) [reply pkgs/docker-images/config/contrail.nix]: No, i'll delete it

File: pkgs/docker-images/default.nix (hunk 2 above)
Comment (k-ayache) [reply pkgs/docker-images/default.nix]: I wanted to add time and format_time in events, yes it's useless, no time field in the event

File: pkgs/docker-images/default.nix (hunk 1 above)
Comment (k-ayache) [reply pkgs/docker-images/default.nix]: Control is built using buildContrailImageWithPerp and he is not concerned

Comment (nlewo) [reply pkgs/docker-images/default.nix]: Or maybe, only if fluentd is enabled. From the user point of view, it is surprising that stderr logs don't go through fluentd.

**Comment by nlewo** [comment]
It seems the fluentd configuration files for all python services are identical. If it is the case, could you create an attribute sets once and reuse it when needed? It seems this is also true for c++ contrail services.

**Pull request merged**